tinygrad/codegen/kernel.py

```diff
@@ -704,12 +704,19 @@ def fix_st(warp_dims, tcd_dims, tcd_expand, pattern_1, pattern_2, st1):
           return LazyOp(op.op, (ret,), new_reduce_axes) if (new_reduce_axes:=tuple(i for i in arg if i-self.first_upcast not in reduce_axes)) else ret
         if self.group_for_reduces:
           start = LazyOp(op.op, tuple(fixup_ast(x, apply_to_st) for x in op.src), arg)
-          local_shape = (1,) * self.global_dims + self.full_shape[self.global_dims:self.global_dims+self.local_dims+self.group_for_reduces] + \
-            (1,) * (self.first_upcast - self.group_for_reduces - self.first_reduce) + tuple([x[0] for x in self.upcasted_axis(0)])
-          local_buffer = MemBuffer(-1, start.dtype, ShapeTracker.from_shape(local_shape))
+          arg = tuple(i for i in range(self.first_reduce, self.first_reduce+self.group_for_reduces) \
+                      if self.sts[reduce_idx].shape[i] != self.sts[reduce_idx+1].shape[i])
+          if len(arg) == 0: return start
```

Comment:
can you add a comment for this? what does 0 axis mean?

Reply:
yes I can, though I want your opinion a bit: this happens if one reduce is grouped and the other is not, thus `group_for_reduces` is true but there's nothing to group for this reduce.

should this be treated as a mistake? the highlighted code can handle it but it should be discouraged

Reply:
hmm, I see this only happens in indexing kernels.
The reduceop in indexing is special, it's an arange we fold into the other reduce (kind of like merging the two reduceops/folding the arange inside the reduceop). Can we just skip group for reduce in these?

Reply:
this can happen if you group one reduce axis and not the other. 

`test_partial_opt_multireduce` in` test_linearizer.py:267` makes sure these cases are correct, but there's no good reason someone would want to do that